openapi/components/schemas/Common/CommonAuthenticationToken.yaml

```
@@ -0,0 +1,23 @@
+type: object
+title: Authentication Token
+properties:
+  token:
+    description: The token identifier string.
+    type: string
+    readOnly: true
+  mode:
+    description: The token's generation mode. A SecretApiKey must be used to login in passwordless mode.
```

nitpick: I found 'mode' confusing here

I think 'type' is a good alternative when describing this field. In my thinking, we currently have two types of token, one which provides access with a password, and one which provides access without.

```suggestion
    description: |-
    Type of token authentication token to generate. The `password` token requires the 
    user to enter a password to log in. The `passwordless` token, requires a secret API 
    key to log in. To obtain an API key, see 
    [Manage API keys](https://www.rebilly.com/docs/dev-docs/api-keys/#manage-api-keys).
```

---

When we say `type` we generally mean the entity becomes or is of this type. In this case `mode` describes only the method of verifying authenticity and has nothing to do with the resulting entity. If `mode: passwordless` is passed for this operation, the authentication method must be a secret key for this exact request. With `mode: password` – the actual password must be supplied in the same request. The token you get from this operation will be the same independently of the `mode`.

---

Excellent input! Thank you 🙏

I've updated based on your comment